### ISSUE TITLE
Cannot reproduce the performance on CasiaB dataset.

### ISSUE THREAD

**Issue by vdhuan** [bb12346/GaitGL#4]
Hi the author,
I tried to reproduce the performance on CasiaB (NM: 97.4 BG: 94.5 CL: 83.6) as presented in the paper: https://openaccess.thecvf.com/content/ICCV2021/papers/Lin_Gait_Recognition_via_Effective_Global-Local_Feature_Representation_and_Local_Temporal_ICCV_2021_paper.pdf 

The config is just the same as reported :
```
  "model": {
      'lr': 1e-4,
      'hard_or_full_trip': 'full',
      'batch_size': (8, 8),
      'restore_iter': 0,
      'total_iter': 80000,
      'margin': 0.2,
      'num_workers': 0,
      'frame_num': 30,
      'hidden_dim':256,
      'model_name': 'GaitGL_CASIAB'
  }
  ```
 changes in GaitGL/model/network/vgg_c3d.py
 line 42:
 ```
      def __init__(self, inplanes, planes, kernel=3, bias=False, p=2, FM=False, **kwargs):
      =>
      def __init__(self, inplanes, planes, kernel=3, bias=False, **p=8**, FM=False, **kwargs): 
 ```
  line 111:
```
      self.conv2dlayer25b_3d = BasicConv3d_p(_set_channels[2], _set_channels[2])
      =>
      self.conv2dlayer25b_3d = BasicConv3d_p(_set_channels[2], _set_channels[2], **FM=True**)
```
line 167-170:
```
         x2da3d = self.conv2dlayer3a_3d(x2d)
        # print('conv2dlayer3a_3d-',x2da3d.shape)
        x2db3d = self.conv2dlayer3b_3d(x2da3d)
        # print('conv2dlayer3b_3d-',x2db3d.shape)
       => 
      x2da3d = x2d
```
line 121: 
```
self.bin_numgl = [32*2]
self.fc_bin = nn.ParameterList([
            nn.Parameter(
                nn.init.xavier_uniform_(
                    torch.zeros(sum(self.bin_numgl), _set_channels[3], _set_channels[3])))
                    ])
=>
self.bin_numgl = [32*2]
        self.fc_bin = nn.ParameterList([
            nn.Parameter(
                nn.init.xavier_uniform_(
                    torch.zeros(sum(self.bin_numgl), _set_channels[**2**], _set_channels[**2**])))
                    ])
```

However, the following is the result I got. 
```
===Rank-1 (Include identical-view cases)===
NM: 96.726,     BG: 91.905,     CL: 80.083
---Rank-1 (Exclude identical view cases)---
NM: 96.471,     BG: 91.196,     CL: 78.482
```

Could you help me out with what is wrong with my config? I cannot reproduce the same performance even run many times.

**Comment by bb12346** [comment]
Hi, vdhuan, your config is correct. However, this is a previous version that just uses the triplet loss for optimization. In the ICCV version, we use a combined loss consisting of a triplet loss and a cross-entropy loss. Thus, you cannot get the same results. I suggest you can use GaitGL from [Opengait](https://github.com/ShiqiYu/OpenGait) to get the same result.

**Comment by vdhuan** [comment]
Oh, I got it now. Many thanks for your quick response.
